actionpack/lib/action_dispatch/routing/inspector.rb

```diff
@@ -82,11 +81,19 @@ def format(formatter, filter = nil)
 
       private
 
-      def filter_routes(filter)
-        if filter
+      def filter_routes(options)
```

Comment:
How about expanding options, like:

```ruby
def filter_routes(filter:, controller:, pattern:)
  filter ||= controller.downcase.sub(/_?controller\z/, '') if controller

  filtered_routes = if filter
    @routes.select { |route| route.defaults[:controller] == filter }
  else
    @routes
  end

  pattern = /#{pattern}/
  filtered_routes.select { |r| r.defaults.each_value.join(' '.freeze) =~ pattern }
end
```

That would get rid of the `filter_routes` method that calls `filter`, which feels weird :smile:

Sorry for backing down on removing the else branch that sets `@routes`, I think it's better the original way now :heart: